WTWSMS/common/buildings/WtWSMS_historical_castle.txt

```diff
@@ -0,0 +1,220 @@
+		potential = {
+			FROMFROM = { title = b_constantinople }
+		}
+		desc = ca_sacred_palace_desc
+		gold_cost = 0
+		build_time = 0
+
+		monthly_character_prestige = 0.3
+
+		ai_creation_factor = 100
+
+		extra_tech_building_start = 0.0
+	}
+	
+	ca_holy_apostles = { # Church of the Holy Apostles
+		potential = {
+			FROMFROM = { title = b_constantinople }
+		}
+		trigger = { TECH_TEMPLE_CONSTRUCTION = 3 }
+		desc = ca_holy_apostles_desc
+		gold_cost = 500
+		build_time = 360
+
+		monthly_character_piety = 0.1
+
+		ai_creation_factor = 100
+
+		extra_tech_building_start = 3.0
+	}
+	
+	ca_hadrian_wall_0 = { # Remnants of Hadrian's Wall
+		potential = { 
+			OR = {
+				FROMFROM = { title = b_burgh }
+				FROMFROM = { title = b_bamburgh }
+			}
+		}
+		desc = ca_hadrian_wall_0_desc
+		gold_cost = 0
+		build_time = 0
+
+		monthly_character_prestige = 0.1
+		fort_level = 0.5
+
+		ai_creation_factor = 100
+
+		extra_tech_building_start = 0.0
+	}
+	
+	ca_antonine_wall_0 = { # Remnants of Antonine Wall
+		potential = { 
+			OR = {
+				FROMFROM = { title = b_dumbarton }
+				FROMFROM = { title = b_stirling }
+			}
+		}
+		desc = ca_antonine_wall_0_desc
+		gold_cost = 0
+		build_time = 0
+
+		monthly_character_prestige = 0.05
+		fort_level = 0.3
+
+		ai_creation_factor = 100
+
+		extra_tech_building_start = 0.0
+	}
+			
+	ca_hadrian_wall_1 = { # Rebuilt Hadrian's Wall
+		potential = { 
+			OR = {
+				FROMFROM = { title = b_burgh }
+				FROMFROM = { title = b_bamburgh }
+			}
+		}
+		trigger = { TECH_CASTLE_CONSTRUCTION = 2 }
+		desc = ca_hadrian_wall_1_desc
+		upgrades_from = ca_hadrian_wall_0
+		gold_cost = 500
+		build_time = 600
+
+		monthly_character_prestige = 0.1
+		fort_level = 0.5
+
+		ai_creation_factor = 5
+
+		extra_tech_building_start = 1.0
+	}
+
+	ca_antonine_wall_1 = { # Rebuilt Antonine Wall
+		potential = { 
+			OR = {
+				FROMFROM = { title = b_dumbarton }
+				FROMFROM = { title = b_stirling }
+			}
+		}
+		trigger = { TECH_CASTLE_CONSTRUCTION = 2 }
+		desc = ca_antonine_wall_1_desc
+		upgrades_from = ca_antonine_wall_0
+		gold_cost = 500
+		build_time = 600
+
+		monthly_character_prestige = 0.05
+		fort_level = 0.3
+
+		ai_creation_factor = 5
+
+		extra_tech_building_start = 1.0
+	}
+		
+	ca_colosseum = { # Colosseum
+		potential = {
+			FROMFROM = { title = b_roma }
+		}
+		desc = ca_colosseum_desc
+		gold_cost = 0
+		build_time = 0
+		convert_to_castle = tp_colosseum
+
+		liege_prestige = 0.3
+		monthly_character_prestige = 0.3
+
+		ai_creation_factor = 100
+
+		extra_tech_building_start = 0.0
+	}
+	
+
+	ca_aurelian_walls = { # Aurelian Walls
+		potential = {
+			FROMFROM = { title = b_roma }
+		}
+		desc = ca_aurelian_walls_desc
+		upgrades_from = ca_wall_5
+		gold_cost = 200
+		build_time = 1460
+		convert_to_castle = tp_aurelian_walls
+		
+		fort_level = 3.0
+		ai_creation_factor = 86
+
+		extra_tech_building_start = 0.0
+	}
+	
+	ca_carthport_0 = { # Ruined Port of Carthage, credits to Morloran
```

Comment:
Ok, I will rename it to Port of Carthage and the upgrade to Expanded Port of Carthage